gapis/api/vulkan/frame_loop.go

```diff
@@ -253,6 +257,24 @@ func (f *frameLoop) Transform(ctx context.Context, cmdId api.CmdID, cmd api.Cmd,
 				// Iterate through the loop contents, emitting instructions one by one.
 				for cmdIndex, cmd := range f.capturedLoopCmds {
 
+					switch cmd.(type) {
+					case *VkUnmapMemory:
+						vkCmd := cmd.(*VkUnmapMemory)
+						vkCmd.Extras().Observations().ApplyReads(out.State().Memory.ApplicationPool())
+						memID := vkCmd.Memory()
```

Comment:
Do we need this ApplyReads? What does it do?

Reply:
My understanding is that when we need to get the value of parameters from the capture, we need to `ApplyRead` or `ApplyWrite` depends on the command.

Reply:
You only need to ApplyRead or ApplyWrite if you are reading memory. (MustRead()) if you are just reading the parameters, you do not have to.